### PR TITLE
docs(api): specify types to reference by link

### PR DESCRIPTION
**Pull request by KaelWD**
before:
Screenshot_20221018_000157
Screenshot_20221018_000344

after:
Screenshot_20221018_000317
Screenshot_20221018_000358

### CODE REVIEW

**Review by nekosaur — APPROVED** [review]

File: packages/api-generator/src/index.ts [hunk 1]
```diff
@@ -18,7 +26,7 @@ const run = async () => {
   const pool = new Piscina({
     filename: './src/worker.js',
     niceIncrement: 10,
-    maxThreads: Math.max(1, Math.floor(Math.min(os.cpus().length / 2, os.freemem() / (1.1 * 1024 ** 3)))),
+    maxThreads: inspector.url() ? 1 : Math.max(1, Math.floor(Math.min(os.cpus().length / 2, os.freemem() / (1.1 * 1024 ** 3)))),
```
Comment (johnleider) [inline packages/api-generator/src/index.ts:29]: is there any value in a comment block that explains what this essentially equates to?

File: packages/api-generator/src/types.ts [hunk 2]
```diff
@@ -290,6 +301,13 @@ function generateDefinition (node: Node<ts.Node>, recursed: string[], project: P
   const tc = project.getTypeChecker()
   type = type ?? node.getType()
 
+  if (type.getAliasSymbol()?.getName() === 'NonNullable') {
```
Comment (johnleider) [inline packages/api-generator/src/types.ts:304]: `getAliasSymbol` is called at least 3 times in this file. Could probably be simplified.

**Review by johnleider — CHANGES_REQUESTED** [review]

File: packages/api-generator/src/index.ts (hunk 1 above)
Comment (KaelWD) [reply packages/api-generator/src/index.ts:29]: What, you can't read that?

File: packages/api-generator/src/types.ts (hunk 2 above)
Comment (KaelWD) [reply packages/api-generator/src/types.ts:304]: Parsing is inherently complicated.

The second one could probably change to `symbol?.getName()`, although idk how that behaves if `type.getSymbol()` is hit.

**Pull request merged**